borgmatic/commands/arguments.py

```diff
@@ -523,6 +569,71 @@ def make_parsers():
         '-h', '--help', action='help', help='Show this help message and exit'
     )
 
+    config_parser = subparsers.add_parser(
+        'config',
+        aliases=SUBPARSER_ALIASES['config'],
+        help='Perform configuration file related operations',
+        description='Perform configuration file related operations',
+        add_help=False,
+    )
+
+    config_group = config_parser.add_argument_group('config arguments')
+    config_group.add_argument('-h', '--help', action='help', help='Show this help message and exit')
+
+    config_subparsers = config_parser.add_subparsers(
+        title='config subcommands',
+        description='Valid subcommands for config',
+        help='Additional help',
+    )
+
+    config_bootstrap_parser = config_subparsers.add_parser(
+        'bootstrap',
+        aliases=SUBPARSER_ALIASES['config_bootstrap'],
+        help='Extract the config files used to create a borgmatic repository',
+        description='Extract just the config files that were used to create a borgmatic repository during the "create" operation',
+        add_help=False,
+    )
+    config_bootstrap_group = config_bootstrap_parser.add_argument_group(
+        'config bootstrap arguments'
+    )
+    config_bootstrap_group.add_argument(
+        '--repository',
+        help='Path of repository to extract config files from',
+        required=True,
+    )
+    config_bootstrap_group.add_argument(
+        '--borgmatic-source-directory',
+        help='Path that stores the config files used to create an archive, and additional source files used for temporary internal state like borgmatic database dumps. Defaults to ~/.borgmatic',
+    )
+    config_bootstrap_group.add_argument(
+        '--archive',
+        help='Name of archive to extract config files from, defaults to "latest"',
+        default='latest',
+    )
+    config_bootstrap_group.add_argument(
+        '--destination',
+        metavar='PATH',
+        dest='destination',
+        help='Directory to extract config files into, defaults to /',
+        default='/',
+    )
+    config_bootstrap_group.add_argument(
+        '--strip-components',
+        type=lambda number: number if number == 'all' else int(number),
+        metavar='NUMBER',
+        help='Number of leading path components to remove from each extracted path or "all" to strip all leading path components. Skip paths with fewer elements',
+    )
+    config_bootstrap_group.add_argument(
+        '--progress',
+        dest='progress',
+        default=False,
+        action='store_true',
+        help='Display progress for each file as it is extracted',
+    )
+    config_bootstrap_group.add_argument(
+        '-h', '--help', action='help', help='Show this help message and exit'
+    )
+
```

Comment:
Something that occurs to me: Did you have a chance to play with [borgmatic's bash completion](https://torsion.org/borgmatic/docs/how-to/set-up-backups/#bash) with these argument changes incorporated? For instance: `borgmatic conf<tab>` or `borgmatic config boot<tab>` or even `borgmatic config bootstrap --prog<tab>`. Might also be good to try just `borgmatic boot<tab>` and see what happens.

If some of these don't work as expected, it'd be good to know. They don't necessarily need to be fixed in this PR (the scope is already large enough), but they should be filed for follow-on work. I might even be able to pick them up after they're filed (if they need to be).

Reply:
I use zsh, will completions work for that? I remember trying it when you mentioned this, but no completions worked for me (even with older versions). Let me try with bash.

Reply:
![image](https://github.com/borgmatic-collective/borgmatic/assets/41837037/86994ee3-1434-4c8b-9c8c-11922925d499)
Works!

![image](https://github.com/borgmatic-collective/borgmatic/assets/41837037/8ae3dea3-94a9-4d79-ad47-db58a2a865af)

It recommended borg as a subcommand too though

Reply:
Awesome! Yeah, completions are only implemented for bash and fish right now. zsh would be a logical next step.

Recommending `borg` after `config` should probably be fixed (not in this PR!), as that would produce an invalid command-line. Feel free to file it as a separate issue.